ethereum/core/src/test/java/org/hyperledger/besu/ethereum/privacy/MultiTenancyPrivacyControllerTest.java

```diff
@@ -0,0 +1,298 @@
+
+    assertThatThrownBy(
+            () -> multiTenancyPrivacyController.sendTransaction(transaction, ENCLAVE_PUBLIC_KEY1))
+        .isInstanceOf(MultiTenancyValidationException.class)
+        .hasMessage("Privacy group must contain the enclave public key");
+
+    verify(privacyController, never()).sendTransaction(transaction, ENCLAVE_PUBLIC_KEY1);
+  }
+
+  @Test
+  public void retrieveTransactionDelegatesToPrivacyController() {
+    final ReceiveResponse delegateRetrieveResponse =
+        new ReceiveResponse(new byte[] {}, PRIVACY_GROUP_ID);
+    when(privacyController.retrieveTransaction(ENCLAVE_KEY, ENCLAVE_PUBLIC_KEY1))
+        .thenReturn(delegateRetrieveResponse);
+
+    final ReceiveResponse multiTenancyRetrieveResponse =
+        multiTenancyPrivacyController.retrieveTransaction(ENCLAVE_KEY, ENCLAVE_PUBLIC_KEY1);
+    assertThat(multiTenancyRetrieveResponse)
+        .isEqualToComparingFieldByField(delegateRetrieveResponse);
+    verify(privacyController).retrieveTransaction(ENCLAVE_KEY, ENCLAVE_PUBLIC_KEY1);
+  }
+
+  @Test
+  public void createPrivacyGroupDelegatesToPrivacyController() {
+    final List<String> addresses = List.of(ENCLAVE_PUBLIC_KEY1, ENCLAVE_PUBLIC_KEY2);
+    final PrivacyGroup delegatePrivacyGroup =
+        new PrivacyGroup(PRIVACY_GROUP_ID, Type.PANTHEON, "name", "description", addresses);
+
+    when(privacyController.createPrivacyGroup(
+            addresses, "name", "description", ENCLAVE_PUBLIC_KEY1))
+        .thenReturn(delegatePrivacyGroup);
+
+    final PrivacyGroup privacyGroup =
+        multiTenancyPrivacyController.createPrivacyGroup(
+            addresses, "name", "description", ENCLAVE_PUBLIC_KEY1);
+    assertThat(privacyGroup).isEqualToComparingFieldByField(delegatePrivacyGroup);
+    verify(privacyController)
+        .createPrivacyGroup(addresses, "name", "description", ENCLAVE_PUBLIC_KEY1);
+  }
+
+  @Test
+  public void deletesPrivacyGroupWhenEnclavePublicKeyInPrivacyGroup() {
+    final PrivacyGroup privacyGroupWithEnclavePublicKey =
+        new PrivacyGroup(
+            "", Type.PANTHEON, "", "", List.of(ENCLAVE_PUBLIC_KEY1, ENCLAVE_PUBLIC_KEY2));
+    when(enclave.retrievePrivacyGroup(PRIVACY_GROUP_ID))
+        .thenReturn(privacyGroupWithEnclavePublicKey);
+    when(privacyController.deletePrivacyGroup(PRIVACY_GROUP_ID, ENCLAVE_PUBLIC_KEY1))
+        .thenReturn(ENCLAVE_PUBLIC_KEY1);
+
+    final String privacyGroupId =
+        multiTenancyPrivacyController.deletePrivacyGroup(PRIVACY_GROUP_ID, ENCLAVE_PUBLIC_KEY1);
+    assertThat(privacyGroupId).isEqualTo(ENCLAVE_PUBLIC_KEY1);
+    verify(privacyController).deletePrivacyGroup(PRIVACY_GROUP_ID, ENCLAVE_PUBLIC_KEY1);
+  }
+
+  @Test
+  public void
+      deletePrivacyGroupFailsWithValidationExceptionWhenPrivacyGroupDoesNotContainEnclavePublicKey() {
+    final PrivacyGroup privacyGroupWithoutEnclavePublicKey =
+        new PrivacyGroup("", Type.PANTHEON, "", "", List.of(ENCLAVE_PUBLIC_KEY2));
```

Comment:
I know it doesn't do anything - but is it worth putting PRIVACY_GROU_ID into the first arg here (just to make things look consistent)

Reply:
done